File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> s1(26,0);
        vector<int> t1(26,0);

        for(char i : s){

            s1[i-'a']++;
        }
        for(char i : t){

            t1[i-'a']++;
        }
        for(int i=0; i<26 ; i++){

            if(s1[i]!=t1[i]){
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        for(String str : strs){
            char[] arr = str.toCharArray();
            Arrays.sort(arr);
            String key = new String(arr);
            map.computeIfAbsent(key, k-> new ArrayList<>()).add(str);
        }
        List<List<String> > result = new ArrayList<>();
        for(List<String> s : map.values()){
            result.add(s);
        }
        return result;
    }
}
